files/scripts/restore_neighbors.py

```diff
@@ -0,0 +1,257 @@
+
+    keys = db.keys(db.CONFIG_DB, 'WARM_RESTART|swss')
+    if keys :
+        value = db.get_all(db.CONFIG_DB, keys[0])
+        if 'enable' in value:
+            if (value['enable']) == 'true':
+                db.close(db.STATE_DB)
+                return True
+    db.close(db.STATE_DB)
+    return False
+
+# check if the system warm reboot is enabled.
+# The restore process only happens when system warm reboot enabled and during the bootup
+def is_system_warm_reboot_enabled():
+    db = swsssdk.SonicV2Connector(host='127.0.0.1')
+    db.connect(db.CONFIG_DB, False)
+
+    keys = db.keys(db.CONFIG_DB, 'WARM_RESTART|system')
+    if keys :
+        value = db.get_all(db.CONFIG_DB, keys[0])
+        if 'enable' in value:
+            if (value['enable']) == 'true':
+                db.close(db.STATE_DB)
+                return True
+    db.close(db.STATE_DB)
+    return False
+
+# Use netlink to set neigh table into kernel
+def set_neigh_in_kernel(ipclass, family, intf_idx, dst_ip, dmac):
+    logging.info('set neighbor entries: family: {}, intf_idx: {}, ip: {}, mac {}'.format(
+    family, intf_idx, dst_ip, dmac))
+
+    if family not in ip_family:
+        return
+
+    family_af_inet = ip_family[family]
+
+    ipclass.neigh('set',
+        family=family_af_inet,
+        dst=dst_ip,
+        lladdr=dmac,
+        ifindex=intf_idx,
+        state=ndmsg.states['reachable'])
+
+# build ARP or NS packets depending on family
+def build_arp_ns_pkt(family, smac, src_ip, dst_ip):
+    if family == 'IPv4':
+        eth = Ether(src=smac, dst='ff:ff:ff:ff:ff:ff')
+        pkt = eth/ARP(op=ARP.who_has, pdst=dst_ip)
+    elif family == 'IPv6':
+        nsma = in6_getnsma(inet_pton(AF_INET6, dst_ip))
+        mcast_dst_ip = inet_ntop(AF_INET6, nsma)
+        dmac = in6_getnsmac(nsma)
+        eth = Ether(src=smac,dst=dmac)
+        ipv6 = IPv6(src=src_ip, dst=mcast_dst_ip)
+        ns = ICMPv6ND_NS(tgt=dst_ip)
+        ns_opt = ICMPv6NDOptSrcLLAddr(lladdr=smac)
+        pkt = eth/ipv6/ns/ns_opt
+    return pkt
+
+# Set the statedb "NEIGH_RESTORE_TABLE|Flags", so neighsyncd can start reconciliation
+def set_statedb_neigh_restore_done():
+    db = swsssdk.SonicV2Connector(host='127.0.0.1')
+    db.connect(db.STATE_DB, False)
+    db.set(db.STATE_DB, 'NEIGH_RESTORE_TABLE|Flags', 'restored', 'true')
+    db.close(db.STATE_DB)
+    return
+
+
+def main():
+
+    print "restore_neighbors service is started"
+    # if swss or system warm reboot not enabled, don't run
+    if not is_system_warm_reboot_enabled() and not is_swss_warm_restart_enabled():
+        print "restore_neighbors service is skipped as warm restart not enabled"
+        # make supervisord happy
+        time.sleep(2)
+        return
+
+    if not is_system_warm_reboot_enabled():
+        set_statedb_neigh_restore_done()
+        print "restore_neighbors service is done as system warm reboot not enabled"
+        # make supervisord happy
+        time.sleep(2)
```

Comment:
you do not need this.

you can add in your supervisord group 

startretries=0
startsecs=0

Reply:
will try that out.